vsomeip-sys/build.rs

```diff
@@ -0,0 +1,264 @@
+ ********************************************************************************/
+
+use decompress::ExtractOptsBuilder;
+use reqwest::blocking::Client;
+use std::error::Error;
+use std::fs::File;
+use std::io::{BufRead, BufReader, BufWriter, Write};
+use std::path::{Path, PathBuf};
+use std::process::Command;
+use std::time::Duration;
+use std::{env, fs};
+
+const VSOMEIP_TAGGED_RELEASE_BASE: &str = "https://github.com/COVESA/vsomeip/archive/refs/tags/";
+const VSOMEIP_VERSION_ARCHIVE: &str = "3.4.10.tar.gz";
+
+fn main() -> miette::Result<()> {
+    let out_dir = env::var_os("OUT_DIR").unwrap();
+
+    let vsomeip_interface_path = {
+        // here we allow bringing the user's own vsomeip interface to bind against if they so wish
+        // note that this binding is configured to work with the tagged release above of VSOMEIP_VERSION_ARCHIVE
+        let user_supplied_vsomeip_include_path = env::var("VSOMEIP_INCLUDE_PATH");
+        if let Ok(user_supplied_vsomeip_include_path) = user_supplied_vsomeip_include_path {
+            PathBuf::from(user_supplied_vsomeip_include_path)
+        } else {
+            let vsomeip_archive_dest = Path::new(&out_dir).join("vsomeip").join("vsomeip.tar.gz");
+            let vsomeip_archive_url =
+                format!("{VSOMEIP_TAGGED_RELEASE_BASE}{VSOMEIP_VERSION_ARCHIVE}");
+            let vsomeip_source_folder = Path::new(&out_dir).join("vsomeip").join("vsomeip-src");
+            download_and_write_file(&vsomeip_archive_url, &vsomeip_archive_dest)
+                .expect("Unable to download released archive");
+            decompress::decompress(
+                vsomeip_archive_dest,
+                vsomeip_source_folder.clone(),
+                &ExtractOptsBuilder::default().strip(1).build().unwrap(),
+            )
+            .expect("Unable to extract tar.gz");
+            vsomeip_source_folder.join("interface")
+        }
+    };
+
+    let vsomeip_lib_path = env::var("VSOMEIP_LIB_PATH")
+        .expect("You must supply the path to a vsomeip library install, e.g. /usr/local/lib");
+    let generic_cpp_stdlib = env::var("GENERIC_CPP_STDLIB_PATH")
+        .expect("You must supply the path to generic C++ stdlib, e.g. /usr/include/c++/11");
+    let arch_specific_cpp_stdlib = env::var("ARCH_SPECIFIC_CPP_STDLIB_PATH").expect("You must supply the path to architecture-specific C++ stdlib, e.g. /usr/include/x86_64-linux-gnu/c++/11");
+
+    let project_root = PathBuf::from(env::var("CARGO_MANIFEST_DIR").unwrap());
+    let runtime_wrapper_dir = project_root.join("src/glue/include"); // Update the path as necessary
+
+    // Somewhat useful debugging
+    // println!("cargo:warning=# CARGO_MANIFEST_DIR : {}", project_root.display());
+    // println!("cargo:warning=# OUT_DIR            : {}", out_path.display());
+    // println!("cargo:warning=# vsomeip_interface  : {}", interface_path.display());
+    // println!("cargo:warning=# runtime_wrapper    : {}", runtime_wrapper_dir.display());
```

Comment:
if this is really useful then why is it commented out?

Reply:
Good point. I need to find a better way that doesn't show up as a yellow, scary warning :sweat_smile:

Reply:
I couldn't find a way to make this a non-scary yellow warning, but I find it to be helpful regardless